src/Workspaces/Core/Portable/Storage/SQLite/SQLitePersistentStorage_WriteBatching.cs

```diff
@@ -120,7 +120,8 @@ internal partial class SQLitePersistentStorage
             return;
 
             // Local functions
-            async Task<(Task previousTask, TaskCompletionSource<int> taskCompletionSource)> GetWriteTaskAsync()
+            //[PerformanceSensitive("https://github.com/dotnet/roslyn/issues/36114", OftenCompletesSynchronously = true)]
+            async ValueTask<(Task previousTask, TaskCompletionSource<int> taskCompletionSource)> GetWriteTaskAsync()
```

Comment:
How does this change reduce allocation?

Reply:
➡️ This method often completes synchronously. When an `async` method completes synchronously, the allocation requirements depend on the return type:

* A method that returns `Task` can internally return `Task.CompletedTask`
* A method that returns `Task<bool>` can internally return a cached task for true or false
* A method that returns a reference type can return a cached instance if it's returning `null`
* A method that returns `ValueTask<T>` (a struct type) can return an instance of `ValueTask<T>` that does not allocate a `Task<T>` behind the scenes

By changing the return type from `Task<T>` to `ValueTask<T>`, we eliminated a requirement to allocate a `Task<T>` instance when the method returns synchronously. As indicated by the comment, this method _often_ returns synchronously with the impact of avoiding the allocation indicated in the original comment above. 👍